plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -1,40 +1,181 @@
-                  "when zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEDF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEDF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEPF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEPF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
+from plasmapy.diagnostics.langmuir import (Characteristic,
+                                           swept_probe_analysis)
+
+N = 30  # array length of dummy probe characteristic
+
+current_arr = np.random.rand(N) * u.A
+bias_arr = np.random.rand(N) * u.V
+
+
+class Test__characteristic_errors:
+    r"""Test the Characteristic class constructor in langmuir.py"""
+
+    bias_infarr = np.append(np.random.rand(N - 1), np.inf) * u.V
+    current_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A
+
+    bias_longarr = np.random.rand(N + 1) * u.V
+    current_longarr = np.random.rand(N + 1) * u.A
+
+    current_arr2 = np.random.rand(N) * u.A
+
+    def test_infinite_I(self):
+        r"""Test error upon an infinite current value"""
+
+        with pytest.raises(ValueError):
+            Characteristic(bias_arr, self.current_infarr)
+
+    def test_infinite_U(self):
+        r"""Test error upon an infinite bias value"""
+
+        with pytest.raises(ValueError):
+            Characteristic(self.bias_infarr, current_arr)
+
+    def test_unequal_arrays(self):
+        r"""Test errors upon unequal array lengths"""
+
+        with pytest.raises(ValueError):
+            Characteristic(self.bias_longarr, current_arr)
+
+        with pytest.raises(ValueError):
+            Characteristic(bias_arr, self.current_longarr)
+
+    def test_addition(self):
+        r"""Test addition of characteristic objects"""
+
+        a = Characteristic(bias_arr, current_arr)
+
+        b = Characteristic(bias_arr, self.current_arr2)
+
+        ab_sum = Characteristic(bias_arr, current_arr + self.current_arr2)
+    
+        errStr = (f"Addition of characteristic objects is not behaving as it "
+                  f"should.")
+        assert (a.current + b.current == ab_sum.current).all(), errStr
+
+    def test_subtraction(self):
+        r"""Test addition of characteristic objects"""
+
+        a = Characteristic(bias_arr, current_arr)
+
+        b = Characteristic(bias_arr, self.current_arr2)
+
+        ab_sub = Characteristic(bias_arr, current_arr - self.current_arr2)
+
+        errStr = (f"Subtraction of characteristic objects is not behaving as "
```

Comment:
Is there a way to display the results when the operation is not behaving as it should? It might be useful to add some display methods like `__str__` to `Characteristic`.

Reply:
There is the debugger mode for pytest that launches pdb after a test failure. I don't think we can always hope to write a nice clean string representation for every single test, especially not work large arrays like here.

Reply:
Might be time for me to learn how to use the debugger :smiley_cat:

Reply:
Suggestions on resources for learning?

Reply:
Mostly scattered around the internets but for googling: python debugger, pdb, optionally ipdb or ipython %debug option, some people use pudb so you might like it, pytest post mortem debugging, python post mortem debugging. Sorry I don't have anything more concrete.